hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/BlockStreamTransactionTranslator.java

```diff
@@ -0,0 +1,290 @@
+        }
+        if (txnOutput.hasContractCall()) {
+            trb.contractCallResult(txnOutput.contractCall().contractCallResult());
+        }
+
+        if (txnOutput.hasEthereumCall()) {
+            trb.ethereumHash(txnOutput.ethereumCall().ethereumHash());
+        }
+
+        //            if (txnOutput.hasTopicCreate()) {
+        //                rb.topicID(txnOutput.topicCreate().topicID());
+        //            }
+
+        if (txnOutput.hasSubmitMessage()) {
+            //                rb.topicSequenceNumber(txnOutput.submitMessage().topicSequenceNumber());
+            rb.topicRunningHashVersion(
+                    txnOutput.submitMessage().topicRunningHashVersion().protoOrdinal());
+        }
+
+        //            if (txnOutput.hasCreateToken()) {
+        //                rb.tokenID(txnOutput.createToken().tokenID());
+        //            }
+
+        //            if (txnOutput.hasTokenMint()) {
+        //
+        //            }
+        //            if (txnOutput.hasTokenBurn()) {
+        //
+        //            }
+        //            if (txnOutput.hasTokenWipe()) {
+        //
+        //            }
+
+        if (txnOutput.hasCreateSchedule()) {
+            //                rb.scheduleID(txnOutput.createSchedule().scheduledTransactionId());
+        }
+
+        if (txnOutput.hasCreateSchedule()) {
+            rb.scheduledTransactionID(txnOutput.createSchedule().scheduledTransactionId());
+        }
+        if (txnOutput.hasSignSchedule()) {
+            rb.scheduledTransactionID(txnOutput.signSchedule().scheduledTransactionId());
+        }
+
+        //            if (txnOutput.hasMintToken()) {
+        //                rb.serialNumbers(txnOutput.mintToken().serialNumbers());
+        //            }
+
+        //            if (txnOutput.hasNodeCreate()) {
+        //                rb.nodeId(txnOutput.nodeCreate().nodeID());
+        //            }
+        //            if (txnOutput.hasNodeUpdate()) {
+        //                rb.nodeId(txnOutput.nodeUpdate().nodeID());
+        //            }
+        //            if (txnOutput.hasNodeDelete()) {
+        //                rb.nodeId(txnOutput.nodeDelete().nodeID());
+        //            }
+
+        if (txnOutput.hasUtilPrng()) {
+            final var entropy = txnOutput.utilPrng().entropy();
+            if (entropy.kind() == PRNG_BYTES) {
+                trb.prngBytes(entropy.as());
+            } else if (entropy.kind() == PRNG_NUMBER) {
+                trb.prngNumber(entropy.as());
+            }
+        }
+
+        maybeAssignEvmAddress(trb, txnOutput);
+
+        // TODO: assign `newPendingAirdrops` (if applicable)
+
+        trb.receipt(rb.build());
+
+        return trb;
+    }
+
+    private void maybeAssignEvmAddress(TransactionRecord.Builder trb, TransactionOutput txnOutput) {
+        // Are these the only two places where default EVM addresses are assigned?
```

Comment:
"default" -> "EVM address alias"?  I think that's what you mean.

Reply:
HIP-583 says you can get an EVM address alias via a CrytoTransfer and EthereumTransaction too, if that's relevant here.

Reply:
Added a commented version since there doesn't seem to be an alias available on those transaction types